### PR TITLE
[HYD-503] Add buildkit README

### PR DESCRIPTION
**Pull request by owenthereal**
Add buildkit README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+This repository is a hub for the buildkits for various well-known PostgreSQL extensions. Your contributions are welcome! Feel free to enhance our collection by submitting a PR.
+
+## :star: Contributing a Buildkit
+
+Eager to contribute? Follow these steps to add a new buildkit:
+
+1. Fork this repository
+1. Add a buildkit configuration file in the `buildkit` directory
+1. Build the extension using the command `pgxman build -f buildkit/<extension>.yaml` and ensure the build is successful.
+1. Test the built extension with a PostgreSQL instance to verify its functionality.
```
Comment (owenthereal) [inline README.md:15]: I wish we have `pgxman try` soon

File: README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# PGXMan Buildkit
+
+PGXMan Buildkit is a YAML configuration file that `pgxman` employs to define the build and packaging of a PostgreSQL extension.
+Explore the specification for more details [here](https://github.com/pgxman/pgxman/blob/main/spec/buildkit.md).
```
Comment (wuputah) [inline README.md:4]: some suggestions:
* move the spec to this repo - cant read it in `pgxman`
* provide a simple / short example here in README ?
* provide a complete "documented" (e.g. with YAML comments) example in the spec or in a separate file (most people dont want to actually read the spec 😁 )

Comment (wuputah) [reply README.md:4]: Some examples for reference: 
https://render.com/docs/blueprint-spec
https://devcenter.heroku.com/articles/build-docker-images-heroku-yml
https://devcenter.heroku.com/articles/app-json-schema

Comment (owenthereal) [reply README.md]: This is good feedback. I added the following:

* buildkit spec is moved to this repo
* add a short example to RAEDME
* add a complete example to the spec

File: spec/buildkit.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+license: PostgreSQL
+# Keywords relevant to the extension.
+keywords:
+  - keyword1
+  - keyword2
+# Supported architectures.
+arch:
+  - amd64
+  - arm64
+# Maintainers of the extension.
```
Comment (wuputah) [inline spec/buildkit.md:30]:
```suggestion
# Maintainers of the buildkit.
```

File: spec/buildkit.md [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+
+## `name`
+
+- **Description**: Identifies the name of the extension.
+- **Type**: String
+- **Required**: Yes
+
+## `maintainers`
+
+- **Description**: Lists the individuals or organizations responsible for maintaining the extension.
```
Comment (wuputah) [inline spec/buildkit.md:133]:
```suggestion
- **Description**: Lists the individuals or organizations responsible for maintaining the buildkit.
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# PGXMan Buildkit
+
+PGXMan Buildkit is a YAML configuration file that `pgxman` employs to define the build and packaging of a PostgreSQL extension.
+Explore the specification for more details [here](spec/buildkit.md).
```
Comment (wuputah) [inline README.md:4]:
```suggestion
[Explore the specification for more details](spec/buildkit.md).
```

**Review by wuputah — APPROVED** [review]

**Pull request merged**